docs/architecture/adr-040-storage-and-smt-state-commitments.md

```diff
@@ -0,0 +1,171 @@
+# ADR 040: Storage and SMT State Commitments
+
+## Changelog
+
+- 2020-01-15: Draft
+
+## Status
+
+DRAFT Not Implemented
+
+
+## Abstract
+
+Sparse Merke Tree (SMT) is a version of a Merkle Tree with various storage and performance optimizations. This ADR defines a separation of state commitments from data storage and the SDK transition from IAVL to SMT.
+
+
+## Context
+
+Currently, Cosmos SDK uses IAVL for both state commitments and data storage.
```

Comment:
I would define what state commitments are and how it differs from data storage. It can be concise.

Reply:
Isn't it self explaining? State commitment is a commitment to a state. I can add a link to explain more general commitment schemes.